spec/product_shared_examples.rb

```diff
@@ -1,50 +1,51 @@
 RSpec.shared_examples_for "NonReservationProduct" do |product_type|
-  let(:account) { create(:setup_account) }
+  let(:account) { create(:setup_account, owner: user) }
+  let!(:user) { FactoryGirl.create(:user) }
+  let(:product_type) { product_type }
+  let(:facility) { FactoryGirl.create(:facility) }
+  let(:facility_account) { facility.facility_accounts.create!(FactoryGirl.attributes_for(:facility_account)) }
+  let(:product) do
+    FactoryGirl.create(product_type,
+                       facility: facility,
+                       facility_account: facility_account)
+  end
+  let(:order) { create(:order, account: account, created_by_user: user, user: user) }
+  let(:order_detail) { order.order_details.create(attributes_for(:order_detail, account: account, product: product, quantity: 1)) }
 
   before :each do
     # clear out default price groups so they don't get in the way
-    PriceGroup.all.each(&:delete)
-    @product_type = product_type
-
-    @user = FactoryGirl.create(:user)
-    @facility = FactoryGirl.create(:facility)
-    @facility_account = @facility.facility_accounts.create!(FactoryGirl.attributes_for(:facility_account))
-    @price_group = FactoryGirl.create(:price_group, facility: @facility)
-    @price_group2 = FactoryGirl.create(:price_group, facility: @facility)
-
-    FactoryGirl.create(:user_price_group_member, user: @user, price_group: @price_group)
-    FactoryGirl.create(:user_price_group_member, user: @user, price_group: @price_group2)
-
-    @product = FactoryGirl.create(product_type,
-                                  facility: @facility,
-                                  facility_account: @facility_account)
-    @order = create(:order, account: account, created_by_user: @user, user: @user)
-    @order_detail = @order.order_details.create(attributes_for(:order_detail, account: account, product: @product, quantity: 1))
-
-    create(:account_price_group_member, account: @order.account, price_group: @price_group)
-    create(:account_price_group_member, account: @order.account, price_group: @price_group2)
+    UserPriceGroupMember.delete_all
```

Comment:
Yeah, I'm fine with leaving this as is. Not a huge fan of it, but if it's going to be too big of a hassle to clean up, then not worth it.